engine/src/main/java/net/jqwik/engine/execution/CheckedProperty.java

```diff
@@ -55,7 +55,17 @@ public CheckedProperty(
 	}
 
 	public PropertyCheckResult check(Reporting[] reporting) {
-		PropertyConfiguration effectiveConfiguration = configurationWithEffectiveSeed();
+		PropertyConfiguration effectiveConfiguration;
+		try {
+			effectiveConfiguration = configurationWithEffectiveSeed();
+		} catch (FailOnFixedSeedException failOnFixedSeedException) {
+			return PropertyCheckResult.failed(
+					configuration.getStereotype(), propertyName, 0, 0,
+					configuration.getSeed(), configuration.getGenerationMode(),
+					configuration.getEdgeCasesMode(), 0, 0,
+					null, null, failOnFixedSeedException
+			);
+		}
```

Comment:
This catch block feels funny. I can't merge it with the next though due to it needing the actual `effectiveConfiguration`

Reply:
Cannot see a more elegant solution either atm. Sometimes bad looking code has to stay until s.o. suddenly has an epiphany.